core/src/main/java/io/grpc/transport/ServerTransportListener.java

```diff
@@ -48,4 +49,9 @@
    */
   ServerStreamListener streamCreated(ServerStream stream, String method,
       Metadata.Headers headers);
+
+  /**
+   * The transport completed shutting down. All resources have been released.
+   */
+  void transportTerminated();
```

Comment:
I'm still not sure I fully understand the thought process behind the terms.  Are you saying that `shutdown` and `terminated` mean essentially the same thing except that in the case of `shutdown` it was user-initiated?